libraries/cmake/formula/openssl/CMakeLists.txt

```diff
@@ -38,29 +28,133 @@ function(opensslMain)
         "--with-zlib-include=${thirdparty_zlib_INCLUDE_DIRS}"
         "--with-zlib-lib=${thirdparty_zlib_LIBRARIES}"
         linux-x86_64
-        ${openssl_c_flags}
+        -fPIC
+        --sysroot=${CMAKE_SYSROOT}
+        -l:libunwind.a
+        -lpthread
+    )
 
-    BUILD_COMMAND
-     "${CMAKE_COMMAND}" -E make_directory "${CMAKE_INSTALL_PREFIX}/etc/openssl" &&
+    set(build_command
+      "${CMAKE_COMMAND}" -E make_directory "${CMAKE_INSTALL_PREFIX}/etc/openssl" &&
       make depend &&
       make
+    )
+
+    set(optional_install_command
+      make install
+    )
+
+  elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "Darwin")
+    set(configure_command
+      "${CMAKE_COMMAND}" -E env CC="${CMAKE_C_COMPILER}" AR="${CMAKE_AR}"
+      perl ./Configure "--prefix=${CMAKE_INSTALL_PREFIX}" "--openssldir=${CMAKE_INSTALL_PREFIX}/etc/openssl"
+        no-ssl2
+        no-ssl3
+        no-asm
+        no-shared
+        no-weak-ssl-ciphers
+        zlib-dynamic
+        enable-cms
+        "--with-zlib-include=${thirdparty_zlib_INCLUDE_DIRS}"
+        "--with-zlib-lib=${thirdparty_zlib_LIBRARIES}"
+        darwin64-x86_64-cc
+        enable-ec_nistp_64_gcc_128
+        -fPIC
+    )
+
+    set(build_command
```

Comment:
Can we add a note referencing OpenSSL's documentation about this. Without context this looks weird. :D

Reply:
Maybe we should also add a check in CMake for this path? For folks building on different versions of macOS or if this path / Xcode changes we might run into difficult-to-debug situations.